app/src/main/java/com/wafflestudio/snutt2/data/vacancy_noti/VacancyRepositoryImpl.kt

```diff
@@ -35,7 +34,15 @@ class VacancyRepositoryImpl @Inject constructor(
         storage.firstVacancyVisit.update(false)
     }
 
-    override suspend fun updateVacancyBannerCloseDate() {
-        storage.vacancyBannerCloseDate.update(SimpleDateFormat("yyyy-MM-dd", Locale.getDefault()).format(System.currentTimeMillis()))
+    override suspend fun updateVacancyBannerOpenTime() {
+        val calendar = Calendar.getInstance()
+        calendar.add(Calendar.DAY_OF_YEAR, 1)
+        calendar.set(Calendar.HOUR_OF_DAY, 0)
+        calendar.set(Calendar.MINUTE, 0)
+        calendar.set(Calendar.SECOND, 0)
+        calendar.set(Calendar.MILLISECOND, 0)
+
+        val nextDayMidnightMillis = calendar.timeInMillis
+        storage.vacancyBannerOpenTime.update(nextDayMidnightMillis)
     }
```

Comment:
Java Calendar는 맨날 사람들이 별로라고 쓰지 말라고 하는 라이브러리라고 들었음

```suggestion
    override suspend fun updateVacancyBannerOpenTime() {
        val today = TimeUnit.MILLISECONDS.toDays(System.currentTimeMillis())
        val nextOpenDayInMillis = TimeUnit.DAYS.toMillis(today + 1)
        storage.vacancyBannerOpenTime.update(nextOpenDayInMillis)
    }
```

Reply:
ㅋㅋㅋㅋㅋㅋ아 그래? 찾아보니까 꽤 많은 [문제점](https://sujl95.tistory.com/3)이 있네... 리턴 타입을 예측하기 힘들고, 월 enum이 0 based라 불편하고, mutable하다
저걸로 수정할게요

Reply:
근데 이거 뭔가 다르게 동작하는데... TimeUnit.MILLISECONDS.toDays가 timezone에 영향을 받아? System.currentTimeMillis()는 timezone 상관 없이 현재 기기의 locale(한국) 기준으로 측정되는데, vacancyBannerOpenTime에 저장되는 값이 UTC로 저장되어서 좀 다르게 동작함... 일단 계속 알아보는중

Reply:
상관없지않아?

Reply:
아 KST로 자정에 바뀌는 걸 원하는건가 ㅋㅋㅋ

Reply:
8월 9일 오후 11시로 시간 설정 -> 배너 지우기 -> 8월 10일 오전 12시로 시간 설정 -> 배너가 떠야하는데 뜨지 않는다!